packages/material-ui/src/Link/Link.js

```diff
@@ -0,0 +1,155 @@
+  block: {},
+  // Same reset as ButtonBase.root
+  /* Styles applied to the root element if `component="button"`. */
+  button: {
+    position: 'relative',
+    // Remove grey highlight
+    WebkitTapHighlightColor: 'transparent',
+    backgroundColor: 'transparent', // Reset default value
+    // We disable the focus ring for mouse, touch and keyboard users.
+    outline: 'none',
+    border: 0,
+    margin: 0, // Remove the margin in Safari
+    borderRadius: 0,
+    padding: 0, // Remove the padding in Firefox
+    cursor: 'pointer',
+    userSelect: 'none',
+    verticalAlign: 'middle',
+    '-moz-appearance': 'none', // Reset
+    '-webkit-appearance': 'none', // Reset
+    '&::-moz-focus-inner': {
+      borderStyle: 'none', // Remove Firefox dotted outline.
+    },
+  },
+};
+
+function Link(props) {
+  const {
+    block,
+    children,
+    classes,
+    className: classNameProp,
+    component,
+    TypographyClasses,
+    underline,
+    ...other
+  } = props;
+
+  return (
+    <Typography
+      className={classNames(
+        classes.root,
+        {
+          [classes.block]: block,
+          [classes.button]: component === 'button',
+          [classes.inline]: !block,
+        },
+        classes[`underline${capitalize(underline)}`],
+        classNameProp,
+      )}
+      classes={TypographyClasses}
+      component={component}
+      {...other}
+    >
+      {children}
+    </Typography>
+  );
+}
+
+Link.propTypes = {
+  /**
+   *  Controls whether the link is inline or not. When block is true the link is not inline
+   *  when block is false it is.
+   */
+  block: PropTypes.bool,
+  /**
+   * The content of the link.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Override or extend the styles applied to the component.
+   * See [CSS API](#css-api) below for more details.
+   */
+  classes: PropTypes.object.isRequired,
+  /**
+   * @ignore
+   */
+  className: PropTypes.string,
+  /**
+   * The color of the link.
+   */
+  color: PropTypes.oneOf([
+    'error',
+    'inherit',
+    'primary',
+    'secondary',
+    'textPrimary',
+    'textSecondary',
+  ]),
+  /**
+   * The component used for the root node.
+   * Either a string to use a DOM element or a component.
+   */
+  component: componentPropType,
+  /**
+   *  The target of the link. You can read about this property on the
+   *  [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Attributes)
+   */
+  target: PropTypes.oneOf(['_self', '_blank', '_parent', '_top']),
+  /**
+   * `classes` property applied to the [`Typography`](/api/typography/) element.
+   */
+  TypographyClasses: PropTypes.object,
+  /**
+   *  Controls when the link should have an underline.
+   */
+  underline: PropTypes.oneOf(['none', 'hover', 'always']),
+  /**
+   * Applies the theme typography styles.
+   */
+  variant: PropTypes.string,
+};
+
+Link.defaultProps = {
+  color: 'primary',
+  component: 'a',
+  block: false,
```

Comment:
Maybe we should use an enum. `display?: 'block' | 'line-block';` We can potentially support more values.

Reply:
My only worry is the amount of valid values https://developer.mozilla.org/en-US/docs/Web/CSS/display

Reply:
We can focus on the most useful values.

Reply:
I like how powerful the display property would be but I like the syntax of the inline property better 🤔

Reply:
inline style can be just fine true, or even a Box clone.